include/nlohmann/detail/meta/detected.hpp

```diff
@@ -39,6 +39,9 @@ struct detector<Default, void_t<Op<Args...>>, Op, Args...>
 template<template<class...> class Op, class... Args>
 using is_detected = typename detector<nonesuch, void, Op, Args...>::value_t;
 
+template<template<class...> class Op, class... Args>
+struct is_detected_lazy : is_detected<Op, Args...> { };
```

Comment:
IMHO, That's done way better than in my PR. I like the idea of making the number of additional instantiations as small as possible, and this allows to eliminate unnecessary instantiations. 👍

Reply:
> Just two comments to compare with my PR. The appearance of a new PR made me update mine. 😉

Sorry, I didn't realize there was already a PR for this.